sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -510,32 +510,94 @@ class Analyzer(
         case _: NoSuchTableException =>
           u.failAnalysis(s"Table or view not found: ${u.tableName}")
       }
     }
 
-    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case i @ InsertIntoTable(u: UnresolvedRelation, parts, child, _, _) if child.resolved =>
-        i.copy(table = EliminateSubqueryAliases(lookupTableFromCatalog(u)))
-      case u: UnresolvedRelation =>
-        val table = u.tableIdentifier
-        if (table.database.isDefined && conf.runSQLonFile && !catalog.isTemporaryTable(table) &&
-            (!catalog.databaseExists(table.database.get) || !catalog.tableExists(table))) {
-          // If the database part is specified, and we support running SQL directly on files, and
-          // it's not a temporary view, and the table does not exist, then let's just return the
-          // original UnresolvedRelation. It is possible we are matching a query like "select *
-          // from parquet.`/path/to/query`". The plan will get resolved later.
-          // Note that we are testing (!db_exists || !table_exists) because the catalog throws
-          // an exception from tableExists if the database does not exist.
-          u
-        } else {
-          lookupTableFromCatalog(u)
+    // If the database part is specified, and we support running SQL directly on files, and
+    // it's not a temporary view, and the table does not exist, then let's just return the
+    // original UnresolvedRelation. It is possible we are matching a query like "select *
+    // from parquet.`/path/to/query`". The plan will get resolved later.
+    // Note that we are testing (!db_exists || !table_exists) because the catalog throws
+    // an exception from tableExists if the database does not exist.
+    private def isRunningDirectlyOnFiles(table: TableIdentifier): Boolean = {
+      table.database.isDefined && conf.runSQLonFile && !catalog.isTemporaryTable(table) &&
+        (!catalog.databaseExists(table.database.get) || !catalog.tableExists(table))
+    }
+
+    // Change the default database name if the plan is a view, and transformDown with the new
+    // database name to resolve all UnresolvedRelations and Views.
+    // If the view is defined in a DataSource other than Hive, and the view's child is empty,
+    // set the view's child to a SimpleCatalogRelation, else throw an AnalysisException.
+    def resolveView(plan: LogicalPlan): LogicalPlan = plan match {
+      case view: View =>
+        val desc = view.desc
+        val defaultDatabase = desc.viewDefaultDatabase
+        val unresolvedChild = view.child.getOrElse {
+          if (isDatasourceTable(desc)) {
+            SimpleCatalogRelation(lookupDatabaseName(desc), desc)
+          } else {
+            throw new AnalysisException(s"The child of view '${desc.identifier}' is not defined.")
+          }
         }
+        // Resolve all the UnresolvedRelations and Views in the child.
+        val newChild = unresolvedChild transform {
```

Comment:
Since you mention that we use transformDown, should we just explicitly use transformDown at here?